ipldutil/traverser.go

```diff
@@ -23,6 +23,8 @@ var defaultLinkSystem = cidlink.DefaultLinkSystem()
 
 func defaultVisitor(traversal.Progress, ipld.Node, traversal.VisitReason) error { return nil }
 
+// TODO(mvdan): why not just use context.Cancelled with errors.Is?
```

Comment:
It'd be nice to see if that can be completely resolved by doing some global replacing. Related to this is https://github.com/ipfs/go-graphsync/pull/258 where you can see an original TODO in the diff that was partially resolved.

Reply:
@masih @hannahhoward thoughts? happy to replace all uses with `errors.Is` and `context.Cancelled` directly, if it sounds good.

Reply:
Mental note: the only reason we might not want to use `context.Cancelled` is if our `ContextCancelError` type wasn't equivalent, e.g. if it doesn't always replace `context.Cancelled`. Though, arguably, then it should have a different name to clarify when it's used and when it's not versus the standard error.

Reply:
I think we could? I authorize this refactor! 😛